src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -93,63 +112,881 @@ static ColliderQuadInit D_809AFFD8 = {
+        this->dyna.actor.velocity.y = 12.0f;
+    } else if (this->skelAnime.curFrame > 11.0f) {
+        if (!(this->dyna.actor.bgCheckFlags & 1)) {
+            this->skelAnime.curFrame = 11;
+        } else {
+            EnAm_SpawnEffects(this, globalCtx);
+
+            if (this->dyna.actor.world.rot.y == yawToHome) {
+                this->unk_258 = 0;
+            }
+
+            this->dyna.actor.velocity.y = 0.0f;
+            this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+        }
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->unk_258 == 0) {
+        EnAm_SetupMoveToHome(this);
+    }
+
+    this->dyna.actor.shape.rot.y = this->dyna.actor.world.rot.y;
+}
+
+/**
+ * After reaching the home position, spin back to the starting rotation.
+ */
+void EnAm_RotateToInit(EnAm* this, GlobalContext* globalCtx) {
+    if (this->skelAnime.curFrame == 8.0f) {
+        if ((this->dyna.actor.world.pos.x == this->dyna.actor.home.pos.x) &&
+            (this->dyna.actor.world.pos.z == this->dyna.actor.home.pos.z)) {
+            Math_SmoothStepToS(&this->dyna.actor.world.rot.y, this->dyna.actor.home.rot.y, 1, 0x1F40, 0);
+        }
+        this->unk_258 = 2;
+        this->dyna.actor.velocity.y = 12.0f;
+    } else if (this->skelAnime.curFrame > 11.0f) {
+        if (!(this->dyna.actor.bgCheckFlags & 1)) {
+            this->skelAnime.curFrame = 11;
+        } else {
+            this->unk_258 = 1;
+            EnAm_SpawnEffects(this, globalCtx);
+
+            if (this->dyna.actor.home.rot.y == this->dyna.actor.world.rot.y) {
+                this->unk_258 = 0;
+            }
+
+            this->dyna.actor.velocity.y = 0.0f;
+            this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+        }
+    }
+
+    if (this->unk_258 == 2) {
+        Math_SmoothStepToF(&this->dyna.actor.world.pos.x, this->dyna.actor.home.pos.x, 1.0f, 8.0f, 0.0f);
+        Math_SmoothStepToF(&this->dyna.actor.world.pos.z, this->dyna.actor.home.pos.z, 1.0f, 8.0f, 0.0f);
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->unk_258 == 0) {
+        EnAm_SetupSleep(this);
+    }
+
+    this->dyna.actor.shape.rot.y = this->dyna.actor.world.rot.y;
+}
+
+void EnAm_MoveToHome(EnAm* this, GlobalContext* globalCtx) {
+    s16 yawToHome = Math_Vec3f_Yaw(&this->dyna.actor.world.pos, &this->dyna.actor.home.pos);
+
+    if (this->skelAnime.curFrame == 8.0f) {
+        this->dyna.actor.velocity.y = 12.0f;
+        this->dyna.actor.speedXZ = 6.0f;
+    } else if (this->skelAnime.curFrame > 11.0f) {
+        if (!(this->dyna.actor.bgCheckFlags & 1)) {
+            this->skelAnime.curFrame = 11;
+        } else {
+            Math_SmoothStepToS(&this->dyna.actor.world.rot.y, yawToHome, 1, 0xBB8, 0);
+
+            if (this->dyna.actor.bgCheckFlags & 2) {
+                this->unk_258--;
+            }
+
+            this->dyna.actor.velocity.y = 0.0f;
+            this->dyna.actor.speedXZ = 0.0f;
```

Comment:
chained assign?

Reply:
imo this doesnt improve readability, its not like a vector where the chained assign is obvious. especially with the dyna actor access the lines gets long and its easier to read vertically